cloudencrypt/cache.go

```diff
@@ -0,0 +1,59 @@
+package cloudencrypt
+
+import (
+	"context"
+	"time"
+
+	"github.com/dgraph-io/ristretto/v2"
+)
+
+// CacheEncryptor wraps another Encryptor and adds a caching mechanism.
+type CacheEncryptor struct {
+	encryptor Encryptor
+	cache     *ristretto.Cache[[]byte, []byte]
```

Comment:
You can't put names here... as for the key type, internally ristretto makes an integer out of it anyway so it can be anything and since I use `[]byte` everywhere it's the best option here.